app/models/provider_interface/change_offer_form.rb

```diff
@@ -0,0 +1,49 @@
+module ProviderInterface
+  class ChangeOfferForm
+    include ActiveModel::Model
+    attr_accessor :application_choice, :step, :provider_id, :course_id, :course_option_id
+
+    validates :application_choice, presence: true
+
+    # This is a form for a multi-step process, in which each step depends on the previous ones.
+    # Validation needs to take this into account, e.g. it is fine for course_id to be blank if step == :provider
+    #
+    STEPS = %i(provider course course_option confirm update).freeze
+    validates :step, presence: true, inclusion: { in: STEPS, message: '%{value} is not a valid option' }
+
+    def step_after?(symbol)
```

Comment:
Maybe give it a role-ier name?

```suggestion
    def step_after?(step_symbol)
```

Reply:
Done